### PR TITLE
A.J. Stein's Week 32 Feedback for Model Review

### PR DESCRIPTION
**Pull request by aj-stein-nist**
# Committer Notes

As requested, this is the last call phase for catalog, profile, and related modules. Circling back on this now.

### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers
](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

File: src/metaschema/oscal_metadata_metaschema.xml [hunk 1]
```diff
@@ -54,7 +54,12 @@
                         <enum value="alternate">The link identifies an alternative location or format for this resource. Defined by <a href="https://html.spec.whatwg.org/multipage/links.html#linkTypes">the HTML Living Standard</a></enum>
                         <enum value="predecessor-version">This link identifies a resource containing the predecessor version in the version history. Defined by  <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
                         <enum value="successor-version">This link identifies a resource containing the predecessor version in the version history. Defined by <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
+                        <enum value="version-history">This link identifies a resource containing the version history of this document. Defined by <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
```
Comment (aj-stein-nist) [inline src/metaschema/oscal_metadata_metaschema.xml:57]: Just skimmed RFC 5820. Seems like a minor value add that would let people point to a local or web-hosted VCS tool (gitweb, GitHub, even subversion) and is a non-breaking change.

File: src/metaschema/oscal_metadata_metaschema.xml [hunk 2]
```diff
@@ -54,7 +54,12 @@
...
                     </allowed-values>
+                    <!--
```
Comment (aj-stein-nist) [inline src/metaschema/oscal_metadata_metaschema.xml]: I would like to discuss the feasibility and usefulness of this.

File: src/metaschema/oscal_metadata_metaschema.xml [hunk 3]
```diff
@@ -924,7 +929,7 @@
         <formal-name>OSCAL Version</formal-name>
         <description>The OSCAL model version the document was authored against and will conform to as valid.</description>
         <remarks>
-            <p>Indicates the version of the OSCAL model to which the document conforms, for example <q>1.1.0</q> or <q>1.0.0-M1</q>. That can be used as a hint for a tool indicating which version of the OSCAL XML or JSON schema to use for validation.</p>
+            <p>Indicates the version of the OSCAL model to which the document conforms, for example <q>1.1.0</q> or <q>1.0.0-milestone1</q>. That can be used as a hint for a tool indicating which version of the OSCAL XML or JSON schema to use for validation.</p>
```
Comment (aj-stein-nist) [inline src/metaschema/oscal_metadata_metaschema.xml:929]: Technically, the version numbers we use are not M1 or M3, but it is spelled out like so if we want to be accurate.

File: src/metaschema/oscal_metadata_metaschema.xml (hunk 2 above)
Comment (aj-stein-nist) [reply src/metaschema/oscal_metadata_metaschema.xml]: I was asked to convert this into an issue for future development, it is not seen as an immediate enhancement or fix to improve the maintenance of the current model in this area.

File: src/metaschema/oscal_control-common_metaschema.xml [hunk 4]
```diff
@@ -205,6 +213,14 @@
...
         <define-assembly name="parameter-guideline">
                 <formal-name>Guideline</formal-name>
+                <!-- 
+                        TODO: Is `@ns` flag enough or should we add system for
+                        dictating guidelines?
+
+                        Use cases: RMF recommends one thing, maybe different with
+                        FedRAMP, then a contradiction or subset of constraints is
+                        required, not recommended for example FCEB agency (Important Federal Agency).
+                -->
```
Comment (david-waltermire) [inline src/metaschema/oscal_control-common_metaschema.xml]: @aj-stein-nist  Please create a new issue for this use case.

Comment (david-waltermire) [inline src/metaschema/oscal_control-common_metaschema.xml]:
```suggestion
```

File: src/metaschema/oscal_control-common_metaschema.xml [hunk 5]
```diff
@@ -184,6 +184,14 @@
...
                 <formal-name>Constraint</formal-name>
                 <description>A formal or informal expression of a constraint or test</description>
+                <!-- 
+                        TODO: Is `@ns` flag enough or should we add system for
+                        dictating constraints?
+
+                        Use cases: RMF recommends one thing, maybe different with
+                        FedRAMP, then a contradiction or subset of constraints is
+                        required, not recommended for example FCEB agency (Important Federal Agency).
+                -->
```
Comment (david-waltermire) [inline src/metaschema/oscal_control-common_metaschema.xml]: @aj-stein-nist Please create a new issue for this use case.

```suggestion
```

File: src/metaschema/oscal_metadata_metaschema.xml (hunk 1 above)
Comment (david-waltermire) [reply src/metaschema/oscal_metadata_metaschema.xml:57]: Ok. This is fine.

File: src/metaschema/oscal_control-common_metaschema.xml (hunk 4 above)
Comment (aj-stein-nist) [reply src/metaschema/oscal_control-common_metaschema.xml]: Sounds good to me, added https://github.com/david-waltermire-nist/OSCAL/issues/54 for follow up. Dave will convert this to a usnistgov/OSCAL issue after our review meeting. (Sorry, Dave.)

File: src/metaschema/oscal_metadata_metaschema.xml (hunk 2 above)
Comment (david-waltermire) [reply src/metaschema/oscal_metadata_metaschema.xml]: I think we can only check for if the last-modified is >= the latest revision.

File: src/metaschema/oscal_metadata_metaschema.xml [hunk 6]
```diff
@@ -54,7 +54,12 @@
...
                         <enum value="predecessor-version">This link identifies a resource containing the predecessor version in the version history. Defined by  <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
                         <enum value="successor-version">This link identifies a resource containing the predecessor version in the version history. Defined by <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
+                        <enum value="version-history">This link identifies a resource containing the version history of this document. Defined by <a href="https://tools.ietf.org/html/rfc5829">RFC 5829</a>.</enum>
                     </allowed-values>
+                    <!-- 
+                        TODO: Can we and should we add a constraint to say, if one revision exists or at least one revision of n>1 exist,
+                        the metadata/published and metadata/last-modified should match?
+                    -->
```
Comment (david-waltermire) [inline src/metaschema/oscal_metadata_metaschema.xml]:
```suggestion
                    <!-- TODO: check for if the last-modified is >= the latest revision. -->
```

File: src/metaschema/oscal_metadata_metaschema.xml (hunk 3 above)
Comment (david-waltermire) [reply src/metaschema/oscal_metadata_metaschema.xml:929]: yep. Good catch!

File: src/metaschema/oscal_control-common_metaschema.xml (hunk 4 above)
Comment (david-waltermire) [reply src/metaschema/oscal_control-common_metaschema.xml]: See #54.

Comment (aj-stein-nist) [reply src/metaschema/oscal_control-common_metaschema.xml]: Done, follow-up issue is https://github.com/usnistgov/OSCAL/issues/1410.

**Pull request merged**